benches/index_build_benchmark.rs

```diff
@@ -1,14 +1,37 @@
 use arklib::index::ResourceIndex;
-use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use criterion::{
+    black_box, criterion_group, criterion_main, BenchmarkId, Criterion,
+};
 
 fn index_build_benchmark(c: &mut Criterion) {
-    let path = "tests/"; // Set the path to the directory containing the resources
+    let path = "tests/"; // Set the path to the directory containing the resources here
 
-    c.bench_function("index_build", move |b| {
-        b.iter(|| {
-            let _index = ResourceIndex::build(black_box(path.to_string()));
-        });
-    });
+    // assert the path exists and is a directory
+    assert!(
+        std::path::Path::new(path).is_dir(),
+        "The path: {} does not exist or is not a directory",
+        path
+    );
+
+    let mut group = c.benchmark_group("index_build");
+    group.measurement_time(std::time::Duration::from_secs(20)); // Set the measurement time here
+
+    group.bench_with_input(
+        BenchmarkId::new("index_build", path),
+        &path,
+        |b, path| {
+            b.iter(|| {
+                ResourceIndex::build(black_box(path.to_string()));
+            });
+        },
+    );
+
+    group.finish();
+
+    // Print the number of collisions (i.e. resources with the same hash)
+    let index = ResourceIndex::build(path.to_string());
```

Comment:
Let's reuse the same index instance from the benchmark runs.

This way, the control of runs number would be more precise.. Imagine huge folder on local filesystem.

Reply:
good point!